server/src-lib/Hasura/RQL/DDL/Metadata.hs

```diff
@@ -419,6 +453,19 @@ fetchMetadata = do
                 FROM hdb_catalog.hdb_function
                 WHERE is_system_defined = 'false'
                     |] () False
+    joinRemoteSchemaMeta schemas perms =
+      flip map schemas $ \(AddRemoteSchemaQuery name def comment) ->
+        RemoteSchemaMeta
+          name
+          def
+          comment
+          (Just
+             (mapMaybe
+                (\(RemoteSchemaPermissions name' role permDef) ->
+                   if name == name'
+                     then Just (RemoteSchemaPermMeta role permDef)
+                     else Nothing)
```

Comment:
We can totally avoid this if do joining in SQL itself

Reply:
See discussion here: https://github.com/hasura/graphql-engine/pull/2690#discussion_r321581083